ansible/roles/block_devices/README.md

```diff
@@ -1,17 +1,19 @@
 block_devices
 =============
 
-Manage filesystems on block devices, including creating partitions, creating filesystems and mounting filesystems.
+Manage filesystems on block devices (such as OpenStack volumes), including creating partitions, creating filesystems and mounting filesystems.
 
 This is a convenience wrapper around the ansible modules:
 - community.general.parted
 - community.general.filesystem
 - ansible.buildin.file
 - ansible.posix.mount
 
-It includes logic to handle OpenStack-provided volumes appropriately both for appliance instances and the Packer build VM.
+To avoid issues with device names changing after e.g. reboots, devices are identified by serial number and mounted by filesystem UUID.
```

Comment:
Is this true still? `control.userdata.tpl` doesn't have any UUIDs in it.

Reply:
I haven't seen swaps after reboot in the same way that I did with the original block_devices role (the above isn't actually a change, just a clarification in the docs). I think part of the problem originally was doing explicit volume attachments in TF, i.e. post-boot. They were definitely liable to change ordering. The TF used here creates the volumes by specifying multiple ephemeral disks for the server, which is more tedious wrt syntax but seems to ensure the device ordering follows the definition ordering. The other part is mounts; looking a node created this way it appears the filesystem gets a label and that is used for the mount, which is safe:

```
[rocky@offboardstate-control ~]$ lsblk -o NAME,TYPE,LABEL,FSTYPE,MOUNTPOINT
NAME   TYPE LABEL FSTYPE MOUNTPOINT
vda    disk              
└─vda1 part       xfs    /
vdb    disk state ext4   /var/lib/state
vdc    disk home  ext4   /exports/home
[rocky@offboardstate-control ~]$ cat /etc/fstab 

#
# /etc/fstab
# Created by anaconda on Sat Jul  2 12:41:01 2022
#
# Accessible filesystems, by reference, are maintained under '/dev/disk/'.
# See man pages fstab(5), findfs(8), mount(8) and/or blkid(8) for more info.
#
# After editing this file, run 'systemctl daemon-reload' to update systemd
# units generated from this file.
#
UUID=1cff4c58-02ed-41c1-b5fc-b29f12aae17a /                       xfs     defaults        0 0
LABEL=state     /var/lib/state  auto    defaults,nofail,x-systemd.requires=cloud-init.service,comment=cloudconfig     02
LABEL=home      /exports/home   auto    defaults,nofail,x-systemd.requires=cloud-init.service,comment=cloudconfig     02
offboardstate-control:/exports/home /home nfs defaults 0 0
```

Reply:
This was more a comment on the docs should probably `s/UUID/LABEL` here.

Reply:
In `block_devices` they are mounted by UUUID: https://github.com/stackhpc/ansible-slurm-appliance/blob/main/ansible/roles/block_devices/tasks/main.yml#L47